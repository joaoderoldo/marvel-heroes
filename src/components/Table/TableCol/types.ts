export interface TableColProps {
  children: React.ReactNode;
  classes?: string | string[];
  dataLabel?: string;
}
