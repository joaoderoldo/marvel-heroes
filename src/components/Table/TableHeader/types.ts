export interface TableHeaderProps {
  children: React.ReactNode;
  classes?: string | string[];
}
