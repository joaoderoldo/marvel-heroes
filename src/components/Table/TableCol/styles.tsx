import styled from "styled-components";

export const TableColWrapper = styled.div`
  width: calc(100% / 3);
  display: flex;
  align-items: center;

  &:last-child {
    justify-content: flex-end;
  }

  @media all and (max-width: 767px) {
    display: flex;
    padding: 10px 0;
    width: 100%;

    &:last-child {
      justify-content: flex-start;
    }

    &:before {
      color: #6c7a89;
      padding-right: 10px;
      content: attr(data-label);
      width: 40%;
      display: flex;
      flex-shrink: 0;
    }
  }
`;
