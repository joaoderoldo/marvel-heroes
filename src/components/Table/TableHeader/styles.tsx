import styled from "styled-components";

export const TableHeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;

  @media all and (max-width: 767px) {
    display: none;
  }

  & > div {
    font-size: 1.25rem;
    padding: 0 24px;
    color: var(--secondary, #ccc);
  }
`;
