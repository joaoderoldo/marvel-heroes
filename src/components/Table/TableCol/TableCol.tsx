import clsx from "clsx";
import { FC } from "react";

import { TableColWrapper } from "./styles";
import { TableColProps } from "./types";

const TableCol: FC<TableColProps> = ({ children, classes, dataLabel }) => {
  return (
    <TableColWrapper className={clsx(classes)} data-label={dataLabel}>
      {children}
    </TableColWrapper>
  );
};

export default TableCol;
