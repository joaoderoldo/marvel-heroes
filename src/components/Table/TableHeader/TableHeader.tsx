import { TableHeaderWrapper } from "./styles";

const TableHeader = ({ children, ...props }) => {
  return <TableHeaderWrapper {...props}>{children}</TableHeaderWrapper>;
};

export default TableHeader;
