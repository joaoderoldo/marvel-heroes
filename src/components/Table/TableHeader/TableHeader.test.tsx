import { render } from "@testing-library/react";
import TableHeader from "./TableHeader";
import TableCol from "../TableCol/TableCol";

const TableHeaderMock = () => (
  <TableHeader classes="table-header">
    <TableCol>Personagem</TableCol>
    <TableCol>Séries</TableCol>
    <TableCol>Eventos</TableCol>
  </TableHeader>
);

describe("TableHeader", () => {
  it("should render the component", () => {
    const { container } = render(<TableHeaderMock />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const { getByText } = render(<TableHeaderMock />);
    expect(getByText("Personagem")).toBeInTheDocument();
    expect(getByText("Séries")).toBeInTheDocument();
    expect(getByText("Eventos")).toBeInTheDocument();
  });
});
