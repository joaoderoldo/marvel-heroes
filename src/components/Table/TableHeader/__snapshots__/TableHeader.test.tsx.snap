// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableHeader > should render the component 1`] = `
<div
  class="sc-beqWaB gVbdkJ table-header"
>
  <div
    class="sc-gueYoa fnNkuk"
  >
    Personagem
  </div>
  <div
    class="sc-gueYoa fnNkuk"
  >
    SÃ©ries
  </div>
  <div
    class="sc-gueYoa fnNkuk"
  >
    Eventos
  </div>
</div>
`;
